packages/amplify-category-function/src/events/prePushHandler.ts

```diff
@@ -11,8 +12,21 @@ import { ensureEnvironmentVariableValues } from '../provider-utils/awscloudforma
  * prePush Handler event for function category
  */
 export const prePushHandler = async (context: $TSContext): Promise<void> => {
-  await ensureEnvironmentVariableValues(context);
-  await ensureFunctionSecrets(context);
+  // if appId and envName can be resolved, proceed with checking env vars and secrets
+  const envName = stateManager.getCurrentEnvName() || context?.exeInfo?.inputParams?.amplify?.envName;
+  // get appId from amplify-meta or fallback to input params
+  const appId: string | undefined =
+    (stateManager.getMeta(undefined, { throwIfNotExist: false }) || {})?.providers?.awscloudformation?.AmplifyAppId ||
+    context?.exeInfo?.inputParams?.amplify?.appId;
```

Comment:
should this be the other way round? I.e. should state/meta file be a fallback here ? Why would we ignore explicit  input params if they're specified ?

If this is valid then other places below might need swap too.

Reply:
There was some existing downstream logic that defaulted to what was in `amplify-meta.json` over input params. This matches that behavior. The existing resolveAppId function in CLI also gives precedence to `amplify-meta.json`: https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-provider-awscloudformation/src/utils/resolve-appId.ts

Reply:
:-)